R/compat-purrr.R

```diff
@@ -7,9 +7,16 @@
 # repository.
 
 map <- function(.x, .f, ...) {
+  .f <- as_function(.f)
   lapply(.x, .f, ...)
 }
+walk <- function(.x, .f, ...) {
+  map(.x, .f, ...)
+  invisible(.x)
+}
+
 map_mold <- function(.x, .f, .mold, ...) {
+  .f <- as_function(.f)
```

Comment:
Should either remove string support or pass an environment to `as_function()`. I would remove string support since I regret that feature and threading the environment is more verbose (need to call `as_function()` at each entry point). In this case we need to pass the `[[` function instead of its name below.

Reply:
I think we're expecting these functions to be used internally, so the default lookup should be ok?

Reply:
I'd rather not encourage function lookup via strings because of the scoping issues involved.

Reply:
The problem is that not allowing strings is going to make the API rather different to both base R and purrr. I think we're less likely to have problems allowing strings with slightly wrong lookup than disallowing them.

Reply:
That said, I don't mind if you want to implement it; but I'm not going to 😄